test/e2e/subroutes_test.go

```diff
@@ -109,66 +110,285 @@ func TestSubrouteVisibilityChange(t *testing.T) {
+	}
 
+	// check route is private
+	if !isRouteClusterLocal(resources.Route.Status) {
+		t.Fatalf("Expected route to be cluster local")
+	}
+
+	// change route - public (Updating ksvc as it will reconcile the route)
+	// check route = public
+	ksvclabelCopy := resources.Service.DeepCopy()
+	labels.SetVisibility(&ksvclabelCopy.ObjectMeta, false)
+	if _, err = v1a1test.PatchService(t, clients, resources.Service, ksvclabelCopy); err != nil {
+		t.Fatalf("Failed to patch service: %s", err.Error())
+	}
+
+	if err = v1a1test.WaitForRouteState(clients.ServingAlphaClient, resources.Route.Name, func(r *v1alpha1.Route) (b bool, e error) {
+		return !isRouteClusterLocal(r.Status), nil
+	}, "Route is public"); err != nil {
+		t.Fatalf("Route is not public: %s", err.Error())
+	}
+
+	publicRoute, err := clients.ServingAlphaClient.Routes.Get(resources.Route.Name, metav1.GetOptions{})
 	if err != nil {
-		t.Fatalf("Timed out waiting for traffic url to change: %s", err.Error())
+		t.Fatalf(err.Error())
+	}
+
+	// check subroute 2 - public
+	if isClusterLocal, err := trafficWithTagIsClusterLocal(publicRoute.Status.Traffic, subrouteTag2); err != nil {
+		t.Fatalf(err.Error())
+	} else if isClusterLocal {
+		t.Fatalf("Expected subroute with tag %s to be not cluster local", subrouteTag2)
+	}
+
+	// check subroute 1 - private  // expect to fail on 0.8.1 (it becomes public)
+	if isClusterLocal, err := trafficWithTagIsClusterLocal(publicRoute.Status.Traffic, subrouteTag1); err != nil {
+		t.Fatalf(err.Error())
+	} else if !isClusterLocal {
+		t.Fatalf("Expected subroute with tag %s to be cluster local", subrouteTag1)
 	}
+
+	//// change subroute 1 - private
+	//// check subroute 1 - private
+	//
+	serviceName1 := fmt.Sprintf("%s-%s", subrouteTag1, resources.Route.Name)
+	svc1, err := clients.KubeClient.Kube.CoreV1().Services(test.ServingNamespace).Get(serviceName1, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get k8s service to modify: %s", err.Error())
+	}
+
+	svc1Copy := svc1.DeepCopy()
+	labels.SetVisibility(&svcCopy.ObjectMeta, true)
+
+	svc1patchBytes, err := test.CreateBytePatch(svc1, svc1Copy)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	if _, err = clients.KubeClient.Kube.CoreV1().Services(test.ServingNamespace).Patch(serviceName1, types.JSONPatchType, svc1patchBytes); err != nil {
+		t.Fatalf("Failed to patch service: %s", err.Error())
+	}
+
+	if err = v1a1test.WaitForRouteState(clients.ServingAlphaClient, resources.Route.Name, v1a1test.IsRouteReady, "Route is ready"); err != nil {
+		t.Fatalf("Route did not become ready: %s", err.Error())
+	}
+
+	if isClusterLocal, err := trafficWithTagIsClusterLocal(publicRoute.Status.Traffic, subrouteTag1); err != nil {
+		t.Fatalf(err.Error())
+	} else if !isClusterLocal {
+		t.Fatalf("Expected subroute with tag %s to be cluster local", subrouteTag1)
+	}
+}
+
+func trafficWithTagIsClusterLocal(tt []v1alpha1.TrafficTarget, tag string) (bool, error) {
```

Comment:
nit: consider a shorter name as per golang best practices. As both the traffic target and tag are taken as arguments it can largely be implied here.